apidef/oas/authentication.go

```diff
@@ -0,0 +1,262 @@
+package oas
+
+import (
+	"reflect"
+
+	"github.com/TykTechnologies/tyk/apidef"
+)
+
+type Authentication struct {
+	Enabled                bool   `bson:"enabled" json:"enabled"` // required
+	StripAuthorizationData bool   `bson:"stripAuthorizationData,omitempty" json:"stripAuthorizationData,omitempty"`
+	Token                  *Token `bson:"token,omitempty" json:"token,omitempty"`
+	JWT                    *JWT   `bson:"jwt,omitempty" json:"jwt,omitempty"`
+}
+
+func (a *Authentication) Fill(api apidef.APIDefinition) {
+	a.Enabled = !api.UseKeylessAccess
+	a.StripAuthorizationData = api.StripAuthData
+
+	if api.AuthConfigs == nil || len(api.AuthConfigs) == 0 {
+		return
+	}
+
+	if authToken, ok := api.AuthConfigs["authToken"]; ok {
+		if a.Token == nil {
+			a.Token = &Token{}
+		}
+
+		a.Token.Fill(api.UseStandardAuth, authToken)
+	}
+
+	if reflect.DeepEqual(a.Token, &Token{}) {
+		a.Token = nil
+	}
+
+	if _, ok := api.AuthConfigs["jwt"]; ok {
+		if a.JWT == nil {
+			a.JWT = &JWT{}
+		}
+
+		a.JWT.Fill(api)
+	}
+
+	if reflect.DeepEqual(a.JWT, &JWT{}) {
+		a.JWT = nil
+	}
+}
+
+func (a *Authentication) ExtractTo(api *apidef.APIDefinition) {
+	api.UseKeylessAccess = !a.Enabled
+	api.StripAuthData = a.StripAuthorizationData
+
+	if a.Token != nil {
+		a.Token.ExtractTo(api)
+	}
+
+	if a.JWT != nil {
+		a.JWT.ExtractTo(api)
+	}
+}
+
+type Token struct {
+	Enabled                 bool `bson:"enabled" json:"enabled"` // required
+	AuthSources             `bson:",inline" json:",inline"`
```

Comment:
is this by intention to miss bson name for this one?

Reply:
Yes